DIS 2.40 (#981)
SkylineCommunications/dataminer-docs: release-notes/DIS/DIS_2.40.md

---
uid: DIS_2.40
---

# DIS 2.40

## New features

### IDE

#### Plugins [ID_34190] [ID_34381] [ID_34391] [ID_34428] [ID_34459]

In the *DIS menu*, you can now find the following plugins. These will allow you to easily add functionality to the *protocol.xml* file you are editing.

| Plugin | Description |
|--------|-------------|
| Plugins \> Add After Startup | Adds the after startup logic to the protocol. DIS will check whether the *protocol.xml* file contains an after startup trigger, and will add one if none was found. Apart from the trigger, it will also add all remaining items of the after startup flow.<br>See also [Executing a QAction after startup](xref:LogicExamples#executing-a-qaction-after-startup) |
| Plugins \> Add matrix... | Adds a matrix and/or Inputs and Outputs tables to the protocol. |
| Plugins \> Add SNMP System Info... | Adds the following SNMP System Info parameters to the protocol:<br>- System Description (1.3.6.1.2.1.1.1)<br>- System Object ID (1.3.6.1.2.1.1.2)<br>- System Uptime (1.3.6.1.2.1.1.3)<br>- System Name (1.3.6.1.2.1.1.5)<br>- System Contact (1.3.6.1.2.1.1.4)<br>- System Location (1.3.6.1.2.1.1.6) |

> | Plugins \> Add SNMP Trap Receiver... | Adds an SNMP trap receiver and a QAction with boilerplate code to process traps to the protocol. |

Reviewer: ```suggestion
| Plugins \> Add SNMP Trap Receiver... | Adds an SNMP trap receiver and a QAction with boilerplate code to process received traps. |
```

Also, it is now possible to override the path associated with a DLL import. To do so, proceed as follows:

1. Right-click the reference (which can be either a reference to a DLL or a reference to a project).
1. Open its properties.
1. Locate the *DLL Path* property, and enter in the value that should override the default path.

#### DIS diagram: Third direction option "Both" [ID_34584]

In the *DIS diagram* window, up to now, after selecting the required depth (i.e. the number of items you want to have displayed starting from or ending with the item you selected), you could select *Forward* or *Reverse* as direction. From now on, you can also select a third option: *Both*.

| Direction | Description |
|-----------|-------------|
| Forward   | Shows the specified\* number of linked items starting from the item you selected. |
| Reverse   | Shows the specified\* number of linked items ending with the item you selected.   |
| Both      | Shows the specified\* number of linked items starting from the item you selected, as well as the specified\* number of linked items ending with the item you selected. |

*\* The value entered in the Depth box.*

> #### XML editor: QAction\@dllImport attribute will now contain the full path to the DLL file [ID_34598]

Reviewer: ```suggestion
#### XML editor: QAction\@dllImport attribute will now include subfolder [ID_34598]
```

In the *QAction\@dllImport* attribute, you can now reference a DLL file that is stored in

- a subfolder of the `Dlls` folder of the repository, or
- a subfolder of the following `C:\Skyline DataMiner` folders: `Files`, `ProtocolScripts` or `ProtocolScripts\DllImport`

> Up to now, the *QAction\@dllImport* attribute would only contain the name of the DLL file. From now on, it will contain the full path to that file.

Reviewer: ```suggestion
Up to now, the *QAction\@dllImport* attribute would only contain the name of the DLL file. From now on, it will include the subfolders a well.
```
